addons/project_mail_plugin/controllers/project_client.py

```diff
@@ -0,0 +1,47 @@
+
+from odoo import http
+from odoo.http import request
+from odoo.tools import html2plaintext, html_sanitize
+
+
+class ProjectClient(http.Controller):
+    @http.route('/mail_plugin/project/search', type='json', auth='outlook', cors="*")
+    def projects_search(self, search_term, limit=5):
+        """
+        Used in the plugin side when searching for projects.
+        Fetches projects that have names containing the search_term.
+        """
+        projects = request.env['project.project'].search([('name', 'ilike', search_term)], limit=limit)
+
+        return [
+            {
+                'project_id': project.id,
+                'name': project.name,
+            }
+            for project in projects
+        ]
+
+    @http.route('/mail_plugin/task/create', type='json', auth='outlook', cors="*")
+    def task_create(self, email_subject, email_body, project_id, partner_id):
+
+        partner = request.env['res.partner'].browse(partner_id).exists()
+        if not partner:
+            return {'error': 'partner_not_found'}
+
+        if not request.env['project.project'].browse(project_id).exists():
+            return {'error': 'project_not_found'}
```

Comment:
No need for a record in the translation file to handle the error message on the Outlook file?

Reply:
The plugins right now do not display the error message directly on the UI, for these errors, a generic message is displayed saying that an error has happened, this message is rather here for debugging purposes, so I don't think that a translation is needed, if it is the case however, we can declare the message only in python using _().

Reply:
Then a generic error message is fine, was just making sure you did not forget it or something :)